pkg/vmprovider/providers/vsphere2/network/network.go

```diff
@@ -518,6 +522,161 @@ func ncpNetIfToResult(
 	return result, nil
 }
 
+// VPCCRName returns the name to be used for the VPC SubnetPort CR.
+func VPCCRName(vmName, networkName, interfaceName string) string {
+	var name string
+
+	if networkName != "" {
+		name = fmt.Sprintf("%s-%s-%s", vmName, networkName, interfaceName)
+	} else {
+		name = fmt.Sprintf("%s-%s", vmName, interfaceName)
+	}
+
+	return name
+}
+
+func createVPCNetworkInterface(
+	vmCtx context.VirtualMachineContextA2,
+	client ctrlruntime.Client,
+	vimClient *vim25.Client,
+	clusterMoRef *vimtypes.ManagedObjectReference,
+	interfaceSpec *vmopv1.VirtualMachineNetworkInterfaceSpec) (*NetworkInterfaceResult, error) {
+
+	kind := interfaceSpec.Network.Kind
+	if kind != "" && kind != "SubnetSet" && kind != "Subnet" {
+		return nil, fmt.Errorf("network kind %q is not supported for VPC", kind)
```

Comment:
Move this to be a part of the switch on L558 so we don't repeat what is allowed here